test/-ext-/string/test_too_many_dummy_encodings.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: false
+require 'test/unit'
+require "-test-/string"
+
+class Test_TooManyDummyEncodings < Test::Unit::TestCase
+  def test_exceed_encoding_table_size
+    assert_separately(%w[--disable=gems], "#{<<~"begin;"}\n#{<<~'end;'}")
+    begin;
+      require "-test-/string"
+      begin
+        1_000.times{|i| Bug::String.rb_define_dummy_encoding("R_#{i}") } # now 256 entries
+      rescue EncodingError => e
+        assert_match(/too many encoding/, e.message)
+      else
+        assert false
+      end
```

Comment:
```suggestion
      assert_raise_with_message(EncodingError, /too many encoding/) do
        1_000.times{|i| Bug::String.rb_define_dummy_encoding("R_#{i}") } # now 256 entries
      end
```

But I don't think this test is necessary.

Reply:
Thanks. It's useful to check the error when this happens, and ko1 had such a test in https://github.com/ruby/ruby/pull/7079/files#diff-d573d0e3e36684ece0beeabd7dbf202b3d789d55f5992ebc9ec53d7c2970c366L164 and now we need to move it to use rb_define_dummy_encoding()